docs/api/1.x/synchronisation.rst

```diff
@@ -13,67 +13,129 @@ This section describes the basic aspects of synchronisation using *Kinto*.
-#. ``GET /buckets/default/collections/articles?_since=<timestamp>`` (with header ``If-Match: <timestamp>``)
-#. If response is ``304 Not modified``, done → up to date, nothing to do.
-#. If no ``Next-Page`` response header, done → store ``ETag`` response header value for next
-   synchronisation.
-#. If ``Next-Page`` response header is present → store the ``ETag`` response header value into a variable ``timestamp``
-   and go on to it (it's an url) using a ``If-Match: <timestamp>`` request header.
-#. If response is ``200 OK`` → repeat previous step.
-#. If response is ``412 Precondition Failed``, some changes occured since the last
-   page → Store the ``ETag``response header into ``before``
-#. Fetch and handle the changes using ``GET collections?_since=<timestamp>&_before=<before>``,
-   and store the ``ETag``response header in to ``timestamp``
-#. Go back to step 5 (follow the ``Next-Page``)
+.. note::
 
+    Pagination requests carry every necessary parameter to be reproduced in case
+    of connectivity error.
 
-Apply changes
-=============
+
+Strategy #1 — Oldest first
+--------------------------
+
+The simplest way to obtain the changes is to sort the records by timestamp
+ascending.
+
+We will use ``sort=last_modified`` and ``_since=<timestamp>``:
+
+#. First sync: ``timestamp := 0``
+#. Next sync: ``timestamp := MAX(local_records['last_modified'])``
+#. Fetch ``GET /buckets/<bid>/collections/<cid>/records?_sort=last_modified&_since=<timestamp>``
+#. If response is ``200 OK``, handle the list of remote changes.
+#. If response has ``Next-Page`` header, follow full URL in header.
+#. If list of changes is empty, **done** → up-to-date.
+
+.. image:: ../../images/sync-oldest.svg
+
+If an error occurs during the obtention of pages,
+the synchronisation can be resumed transparently, since it relies on the highest
+timestamp successfully stored locally.
+
+
+Strategy #2 — Newest first
+--------------------------
+
+In order to populate a UI, it might be relevant to obtain the latest changes first.
+
+Syncing newest records first is a bit more complex since changes can occur between
+the obtention of the first and the last pages.
+
+We will use ``sort=-last_modified`` (desc), ``_before`` to omit later changes,
+and ``_since`` to include changes after last sync:
+
+#. First sync: ``timestamp := 0``
+#. Next sync: use ``timestamp`` stored in last successful sync.
+#. Fetch current collection timestamp ``HEAD /buckets/<bid>/collections/<cid>/records``
+   in ``ETag`` response header and store its value in ``start``.
+#. Fetch ``GET /buckets/<bid>/collections/<cid>/records?_sort=-last_modified&_before=<start>&_since=<timestamp>``
+#. If response is ``200 OK``, stack the obtained list of remote changes.
+#. If response has ``Next-Page`` header, follow full URL in header.
+#. If list of changes is empty, **done** → handle the stack of remote changes
+   and update the timestamp: ``timestamp := MAX(local_records['last_modified'])``
```

Comment:
ok, I remember now, sorry. So then it's safe for Kinto. We could add a note saying that for Syncto, this constraint is not true, and records will actually get skipped when using this approach (in my test account, I had hundreds of records with identical timestamp, because the FxSync sets the first sync time as the timestamp for pre-existing bookmarks).